### PR TITLE
Feat/frontend deployment

### PR DESCRIPTION
**Pull request by alexa-perlov**
*Issue #, if available:*

*Description of changes:*
Adding CDK and Terraform to deploy the Streamlit frontend to ECS behind an ALB and CloudFront. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/document_explorer/client_app/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#syntax=docker/dockerfile:1.4
+# Uncomment below for terraform deployment
+#FROM public.ecr.aws/docker/library/python:3.11.9-slim
+
+# Uncomment below for CDK deployment
+FROM python:3.11-slim
```
Comment (scottschreckengaust) [inline samples/document_explorer/client_app/Dockerfile:6]:
```suggestion
FROM public.ecr.aws/docker/library/python:3.11.9-slim
```

File: samples/document_explorer/README.md
Comment (krokoko) [inline samples/document_explorer/README.md:1]: 2 questions:
- Did you run https://www.checkov.io/ on the TF code ?
- Have you tested with multiple browsers ? Some customers have reported issues with Mozilla Firefox, we should add a warning/note in the README about browser(s) used for testing

Comment (alexa-perlov) [reply samples/document_explorer/README.md:1]: I've tested with Firefox, Chrome, and Safari and have not encountered any issues. Running Checkov now and addressing flags

File: samples/document_explorer/terraform-config-frontend/main.tf [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module "serverless-streamlit-app" {
+  source = "aws-ia/serverless-streamlit-app/aws"
```
[the remark this answers is not in the document]
Comment (alexa-perlov) [reply samples/document_explorer/terraform-config-frontend/main.tf:2]: Check: CKV_TF_1: "Ensure Terraform module sources use a commit hash"  --> will switch to using git commit hash as source (more info: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/supply-chain-policies/terraform-policies/ensure-terraform-module-sources-use-git-url-with-commit-hash-revision)

File: samples/document_explorer/README.md
Comment (alexa-perlov) [reply samples/document_explorer/README.md:1]: have run Checkov and added a .checkov.yml file to handle the suppressions

Comment (krokoko) [reply samples/document_explorer/README.md:1]: awesome thanks

**Review by krokoko — DISMISSED** [review]
LGTM

**Review by scottschreckengaust — DISMISSED** [review]

**Review by krokoko — APPROVED** [review]

**Pull request merged**